### ISSUE TITLE
[BUG] Gradient of QPE circuit returns None for Tensorflow

### ISSUE THREAD

**Issue by lillian542** [PennyLaneAI/pennylane#5002]
### Expected behavior

The gradient for the circuit below would return 0.4282 when using Tensorflow, like it does for PyTorch and JAX. Additionally, the manual circuit for comparison would also return 0.4282 (as it does for the other interfaces).

### Actual behavior

It returns `None` for the QPE circuit, and a different result from the other interfaces (0.14273536) for the comparison circuit

### Additional information

The xfail [on this test](https://github.com/PennyLaneAI/pennylane/blob/b6b1f164b33bb9919ead465007a05ea628d8d551/tests/ops/functions/test_iterative_qpe.py#L123) should be removed when this issue is resolved

### Source code

```shell
        import tensorflow as tf

        def grad(f):
            def wrapper(x):
                with tf.GradientTape() as tape:
                    y = f(x)

                return tape.gradient(y, x)

            return wrapper

        dev = qml.device("default.qubit")

        @qml.qnode(dev)
        def circuit(theta):
            meas = qml.iterative_qpe(qml.RZ(theta, wires=[0]), [1], iters=2)
            return qml.expval(meas[0])

        @qml.qnode(dev)
        def manual_circuit(phi):
            qml.Hadamard(wires=[1])
            qml.ctrl(qml.RZ(phi, wires=[0]) ** 2, control=[1])
            qml.Hadamard(wires=[1])
            qml.CNOT(wires=[1, 2])
            qml.CNOT(wires=[2, 1])
            qml.Hadamard(wires=[1])
            qml.ctrl(qml.RZ(phi, wires=[0]), control=[1])
            qml.ctrl(qml.PhaseShift(-np.pi / 2, wires=[1]), control=[2])
            qml.Hadamard(wires=[1])
            qml.CNOT(wires=[1, 3])
            qml.CNOT(wires=[3, 1])

            return qml.expval(qml.Hermitian([[0, 0], [0, 1]], wires=3))

        phi = tf.Variable(1.0)
        assert np.isclose(grad(circuit)(phi), grad(manual_circuit)(phi))
```

### System information

```shell
pl dev
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by albi3ro** [comment]
Solved for parameter shift by adding:
```
    def single_qubit_rot_angles(self):
        return [self.z * val for val in self.base.single_qubit_rot_angles()]
```
to `PowOperation`. Still tracking down the issue for backprop.
